src/platform/intel/ace/lib/pm_runtime.c

```diff
@@ -59,6 +60,40 @@ const struct pm_state_info *pm_policy_next_state(uint8_t cpu, int32_t ticks)
 }
 #endif /* CONFIG_PM_POLICY_CUSTOM */
 
+static inline void ace_pm_runtime_allow_pgcg(bool enable)
+{
+	if (enable) {
+		pm_policy_state_lock_put(PM_STATE_RUNTIME_IDLE, PM_ALL_SUBSTATES);
+	} else {
+		pm_policy_state_lock_get(PM_STATE_RUNTIME_IDLE, PM_ALL_SUBSTATES);
+		/* Disable power gating when preventing */
+		DFDSPBRCP.bootctl[PLATFORM_PRIMARY_CORE_ID].bctl |=
+			DFDSPBRCP_BCTL_WAITIPCG | DFDSPBRCP_BCTL_WAITIPPG;
+	}
+}
+
+void platform_pm_runtime_enable(uint32_t context, uint32_t index)
+{
+	switch (context) {
+	case PM_RUNTIME_DSP:
+		ace_pm_runtime_allow_pgcg(true);
```

Comment:
sorry, I don't think this makes much sense. `ace_pm_runtime_allow_pgcg()` shares ZERO code between `true` and `false` cases. You make those cases into the same function with a parameter, then it's called from two locations - both with that parameter fixed to a constant. Why not just do what has to be done here and in `disable()` below directly?